Extension/src/LanguageServer/client.ts

```diff
@@ -744,6 +745,7 @@ export class DefaultClient implements Client {
                                     let cancelling: boolean = referencesPendingCancellations.length > 0;
                                     referencesPendingCancellations.push({ reject, callback });
                                     if (!cancelling) {
+                                        renamePending = false;
```

Comment:
>renamePending = false; [](start = 40, length = 22)

Could you remove this?  Although it's cancelling, it's also queueing up a new rename, so should not clear the renamePending flag.